sakisandrac/style-stash
# Sc/feature/item form

What I did:
- Created an Itemform component set to it's own route
- Styled the form so that it looks somewhat acceptable
- Form is fully functional- users can upload items to api
- I created an apiCalls file
- User will see error message if both category and image are not provided upon clicking add item button.
- User will see a message when item has been successfully added to data
- The form clears after the add button is clicked.
- The user sees a preview of the image uploaded after they've selected it.

Testing Notes/Extras: 
- run npm i to install uuid
- open localhost:3000/itemform to test out page- upload an image from computer and choose a category. Open the console to see the log if post request was successful. 

BUGS:
- I am using a useEffect hook to run a post request and put newData in the dependencies array. New data is created when the user successfully fills out the form, and hits the add item button. However, the useEffect will still run when the component mounts for the first time, so if you look in the console, the post function is trying to post empty data to the api, and the api is sending back an error. This bug does not effect the functionality of the form but I imagine it will be an issue once we try to render error handling to the user.

Did this break anything?

Type of change

Checklist:

What's next?
- I think the all of the handle on change functions can combined into one upon refactoring.
- Error handling will need to be added
